### PR TITLE
feat(movies): add list and retrieve movies endpoints

### PR DESCRIPTION
**Pull request by nathankhuu**
## What
- Add a `Makefile` to generate scrips to run commands
- Add an `App` struct containing `Config` and `DB` packages
- Add migrations support
- Add endpoints to retrieve and list movies

## Why
- `Makefile` to speed up development process
- Database to store movies and endpoints to return movies for users

### CODE REVIEW

File: cmd/migrations/20190820225830_add_movies.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            ('Doctor Strange', '2016-10-28'),
+            ('Guardians of the Galaxy Vol. 2', '2017-04-28'),
+            ('Spider-Man: Homecoming', '2017-07-07'),
+            ('Thor: Ragnarok', '2017-10-27'),
+            ('Black Panther', '2018-02-09'),
+            ('Avengers: Infinity War – Part 1', '2018-05-04'),
+            ('Ant-Man and the Wasp', '2018-08-03'),
+            ('Captain Marvel', '2019-03-08'),
+            ('Avengers: Infinity War – Part 2', '2019-05-03'),
+            ('Spider-Man: Far From Home', '2019-07-05')
```
Comment (ajpawlicki) [inline cmd/migrations/20190820225830_add_movies.go:34]: Can we add `Batman v Superman: Dawn of Justice`

File: cmd/migrations/20190820225830_add_movies.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        `)
+		return err
+	}
+
+	down := func(db orm.DB) error {
+		_, err := db.Exec("")
+		return err
+	}
```
Comment (ajpawlicki) [inline cmd/migrations/20190820225830_add_movies.go:42]: More serious question: why aren't we doing anything in this `down`. Don't we want a `DELETE FROM movies` or am I missing something?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+.PHONY: db-setup
+db-setup:
+	make db-reset
+	@createuser goyagi_admin
+	@createdb -O goyagi_admin goyagi
+	make db-migrate
+
+.PHONY: db-migrate
+db-migrate:
+	@go run cmd/migrations/*.go migrate
```
Comment (ajpawlicki) [inline Makefile:69]: Does this command run all migrations? How does it know which migrations have been run already?

File: pkg/movies/routes.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+	// RegiserRoutes takes in an application.App struct and we wrap it around an
+	// internal handler struct. This allows us to create a methods over the
+	// handler struct that act as our endpoint handler functions. Doing this
+	// allows us to access our App struct (which contains references to all
+	// objects that need to only be instantiated once) from all of our handlers.
+	// If we ever need to add new dependencies to our App struct, we'll
+	// automatically make them available to all our handlers through this
+	// construct.
+	h := handler{app}
```
Comment (ajpawlicki) [inline pkg/movies/routes.go:20]: Was curious as to why we don't need to import the `handler` pkg then I realized this is all included in the `movies` pkg.

**Review by ajpawlicki — APPROVED** [review]
LGTM. One thing to consider is the size of the PR. PRs with 500+ line changes are harder to digest and review. I personally like to keep them even smaller and limit PRs to < 200 lines. Of course there are exceptions to the rule. I probably would split this PR up into 3 separate ones. One each for the configurations, migrations, and endpoints. Just a suggestion and something to keep in mind.

File: cmd/migrations/20190820225830_add_movies.go (hunk 2 above)
Comment (nathankhuu) [reply cmd/migrations/20190820225830_add_movies.go:42]: Yeah I think it should, but I think this was just a way to add movies quickly so it wasn't really treated like a true migration if that makes sense.

From goyagi: "For the purpose of this tutorial we will be creating a new migration that adds movies to our database. In a real world scenario we wouldn't be doing this."

File: Makefile (hunk 3 above)
Comment (nathankhuu) [reply Makefile:69]: I'm not exactly sure how it knows, but I'll look into it and get back to you

**Comment by nathankhuu** [comment]
Agree about PR length! For the purposes of Goyagi though, I'll keep following the tutorial so I can move through it, but will definitely keep it smaller when I start my onboarding tasks.

File: cmd/migrations/20190820225830_add_movies.go (hunk 1 above)
Comment (shraddha1704) [reply cmd/migrations/20190820225830_add_movies.go:34]: nit ^

**Review by shraddha1704 — APPROVED** [review]

**Pull request merged**